server/clientRoutes.js

```diff
@@ -117,15 +128,78 @@ function clientsRouter(clients, defaultId) {
   });
 
   // get info about a specific client
-  router.get('/:id', (req, res) => {
+  router.get('/:id', async (req, res) => {
+    let configurations;
+    try {
+      configurations = await getConfig(req.params.id);
+    } catch (e) {
+      logger.error(e);
+      if (e.code === 'ENOENT')
+        return res.status(404).send(`Config for "${req.params.id}" not found`);
+      else
+        return res.status(500).send(`There was a problem with your request.`);
+    }
+
     const info = {
-      network: config.str('network'),
-      chain: config.str('chain', 'bitcoin')
+      configs: configurations.data
     };
+
+    if (req.query.checkStatus) {
+      logger.info(`Checking status of client "${req.params.id}"...`);
+      try {
+        await testConfigOptions(configurations);
+        info.status = true;
+      } catch (e) {
```

Comment:
I don't love `if/else` blocks in a `catch`, but sometimes there is no better way of doing it

Reply:
Totally agree! I was playing with this for a while. Not just `if/else` in catch, but the `try/catch` is itself in an `if` statement 😬. open to suggestions but yeah this felt like the only way given we're looking for a specific case and kind of error.